src/main/java/frc/robot/subsystems/Arm/ArmSubsystem.java

```diff
@@ -0,0 +1,340 @@
+          } else {
+            finalState = ArmState.FLOOR_CONE;
+          }
+        }
+        useFinal = true;
+      }
+    } else {logger.info("cannot pickup nonexistent game piece");}
+  }
+
+  public void yoshi(gamePiece gamePiece) {
+    if (gamePiece == ) {
+      currState = ArmState.PARALLEL_TRANS;
+      desiredState = ArmState.YOSHIED;
+      extendoSubsystem.setPos(ArmConstants.kYoshiPos);
+      finalState = ArmState.FLOOR_CUBE;
+    }
+    if (currState == ArmState.FLOOR_CONE) {
+      currState = ArmState.PARALLEL_TRANS;
+      desiredState = ArmState.YOSHIED;
+      extendoSubsystem.setPos(ArmConstants.kYoshiPos);
+      finalState = ArmState.FLOOR_CONE;
+    }
+    if (currState == ArmState.FLOOR_CONE_UPRIGHT) {
+      currState = ArmState.PARALLEL_TRANS;
+      desiredState = ArmState.YOSHIED;
+      extendoSubsystem.setPos(ArmConstants.kYoshiPos);
+      finalState = ArmState.FLOOR_CONE_UPRIGHT;
+    }
+  }
+
+  public boolean isFinished() {
+    if (shoulderSubsystem.isFinished()
+        && extendoSubsystem.isFinished()
+        && wristSubsystem.isFinished()) {
+      return true;
+    }
+    return false;
+  }
+
+  @Override
+  public void periodic() {
+    if (this.isFinished()) {
+      currState = desiredState;
+      if (useFinal) {
+        switch (finalState) {
+          case HIGH_CUBE:
+            useFinal = false;
+            high(gamePiece.CUBE);
+            break;
+          case HIGH_CONE:
+            useFinal = false;
+            high(gamePiece.CONE);
+            break;
+          case MID_CUBE:
+            useFinal = false;
+            mid(gamePiece.CUBE);
+            break;
+          case MID_CONE:
+            useFinal = false;
+            mid(gamePiece.CONE);
+            break;
+          case LOW_CUBE:
+            useFinal = false;
+            low(gamePiece.CUBE);
+            break;
+          case LOW_CONE:
+            useFinal = false;
+            low(gamePiece.CONE);
+            break;
+          case SHELF_CUBE:
+            useFinal = false;
+            shelf(gamePiece.CUBE);
+            break;
+          case SHELF_CONE:
+            useFinal = false;
+            shelf(gamePiece.CONE);
+            break;
+          case FLOOR_CUBE:
+            useFinal = false;
+            floor(gamePiece.CUBE, true);
+            break;
+          case FLOOR_CONE:
+            useFinal = false;
+            floor(gamePiece.CONE, false);
+            break;
+          case FLOOR_CONE_UPRIGHT:
+            useFinal = false;
+            floor(gamePiece.CONE, true);
+            break;
+          case YOSHIED:
+            // fall through
+          case PARALLEL_TRANS:
+            // fall through
+          case STOW_TO_HIGH:
+            // fall through
+          case STOW:
+            // fall through
+          default:
+            useFinal = false;
+            break;
+        }
+      }
+    }
+
+    if (currState == ArmState.STOW_TO_HIGH
```

Comment:
move this logic before isFinished check so that we don't get stopped mid-motion
also add a similar state for STOW_TO_MID